### ISSUE TITLE
Create with TabbedForm and CustomField not rendering  

### ISSUE THREAD

**Issue by prakashjais99** [marmelab/admin-on-rest#1085]
Custom field accepts props bases state to set its values. But even though create rendering is coming still Create element but not reaching custom field. Was working fine with 1.2.2 version. Not working with latest version.
Below is the code snippet
```
          render() {    
            return(<Create>
                  <TabbedForm>
                    <FormTab label="Order fields">
                        <ExtraFieldComponent
                            source="fields"
                            newInfo={this.state.defaultRecord.fields}/>
                    </FormTab>
                </TabbedForm>
            </Create>);
}

```
ExtafielsComponent componentWillReceiveProps() is never called even though this.state.defaultRecord.fields changed

**Comment by djhi** [comment]
Can you post the full component ?

**Comment by prakashjais99** [comment]
`class ProductCreate extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            defaultRecord: {}
        }
    }


    loadTemplate(values) {
        this.setState({defaultRecord: values.data})
    }

    render() {
        return (
            <Create
                {...this.props}
                title="Create a New product"
                actions={< CreateToolbar {
                ...this.props
            }
            getData = {
                this
                    .getData
                    .bind(this)
            } />}>
                <TabbedForm ref="createProduct" validate={validateCreate} redirect="list">
                    <FormTab label="Basic Info">
                        <TextInput
                            label='Name'
                            source="name"
                            options={{
                            fullWidth: true
                        }}
                            validate={[required]}/>

                    </FormTab>

                    <FormTab label="Extra Fields">
                        <ExtraFieldComponent
                            source="extrafields"
                            defaultValue={this.state.defaultRecord.extrafields}/>
                    </FormTab>
                </TabbedForm>
            </Create>
        );
    }
}`

CreateToolBar has a button with open dialog and when user choose a template that data will be passed back to create component to set

`class LoadTemplate extends Component {

    constructor(props) {
        super(props);
        this.state = {
            value: undefined,
            templates: [],
            selectedValue: undefined
        };
    }

    componentWillMount() {
        restClient(GET_LIST, "Template", {}).then(categories => {
            this.setState({templates: categories.data})
        }, error => {});
    }

    handleChange(event, index, value) {
        log.info(value);
        this.setState({value: value, selectedValue: this.state.templates[index]});
    }

    onLoadClicked(event) {
        this
            .props
            .onTemplateSelected(this.state.selectedValue)
        this
            .props
            .handleClose();
    }

    render() {
        let noTemplates = this.state.templates.length === 0;
        return (
            <Dialog title="Select a template to load" modal={true} open={this.props.open}>
                <div >
                    {noTemplates
                        ? <h4>There are no templates availbale. Save few of them before using load</h4>
                        : <SelectField
                            floatingLabelText="Templates"
                            value={this.state.value}
                            onChange={this
                            .handleChange
                            .bind(this)}
                            style={styles.formdivStyle}>
                            {this
                                .state
                                .templates
                                .map(template => <MenuItem key={template.id} value={template.name} primaryText={template.name}/>)}

                        </SelectField>}
                    <div style={styles.divStyle}>
                        <span style={styles.submitStyle}>
                            <RaisedButton
                                style={styles.buttonStyle}
                                label="Load"
                                primary={true}
                                onClick={this
                                .onLoadClicked
                                .bind(this)}
                                disabled={!this.state || !this.state.selectedValue}/>
                            <RaisedButton
                                style={styles.buttonStyle}
                                secondary={true}
                                label="Cancel"
                                onClick={this.props.handleClose}/>
                        </span>
                    </div>
                </div>
            </Dialog>

        );
    }
}

`

**Comment by djhi** [comment]
Thanks for reporting, I can confirm it's a bug.

The problem is that we need to dispatch actions for handling default values but we only did it in [`componentDidMount`](https://github.com/marmelab/admin-on-rest/blob/master/src/mui/form/FormField.js#L18). We must handle `componentWillReceiveProps` too

**Comment by prakashjais99** [comment]
@djhi 
How to fix it in the current code I tried to add below code in FormField.js. but didn't work
```
    componentWillReceiveProps(nextProps) {
        if (nextProps.input.props.defaultValue) {
            let defaultValue = nextProps.input.props.defaultValue;

            if (typeof defaultValue === 'function') {
                defaultValue = nextProps
                    .input
                    .props
                    .defaultValue();
            }

            nextProps.initializeForm({
                [nextProps.input.props.source]: defaultValue
            });
        }
    }
```

**Comment by zheynov** [comment]
Hello! 
I have the same issue with TabbedForm. Are you going to add a fix to 1.3.4 or later?
Thanks.
